Affirm/Magento2_Affirm
# GraphQl : fix save country_code in sales_order_payment

### Description

Today, if I make a payment in graphQl for the country Canada, it is US code that is used since in graphQl the `country_code` option is not saved in `sales_order_payment.additional_data` database and therefore it is the `US` code that is sent in the `country-code` header.

When calling the Affirm API https://api.global-sandbox.affirm.com/api/v2/checkout/XXX, a 404 error is returned.

![Capture_d_écran_du_2023-10-12_09-22-43](https://github.com/user-attachments/assets/cb3fc271-7a85-4c1d-8403-82b0ee2edaad)

![Capture d’écran du 2024-12-13 14-53-20](https://github.com/user-attachments/assets/d91f4eaa-6598-4e33-a3a4-afc77f0fc623)

This MR saves the `country_code` in the payment when a checkout is done in GraphQl. This MR takes the logic of the `\Astound\Affirm\Model\Checkout::initToken` function used in a classic Magento checkout.

![Capture d’écran du 2024-12-13 14-51-05](https://github.com/user-attachments/assets/03dc3d8e-7681-43ec-8bb5-7f1573790592)

### How To Reproduce

1. Setup Canada region in Affirm Magento configuration
2. Start new checkout in graphQl
3. Place order with Affirm
4. Confirm payment in Affirm UI
5. :boom: 

### Expected behavior

If user place order with CA currency, the header `country-code` must be CAN.
